presto-parquet/src/main/java/io/prestosql/parquet/writer/valuewriter/TimeValueWriter.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.parquet.writer.valuewriter;
+
+import io.prestosql.spi.block.Block;
+import io.prestosql.spi.type.Timestamps;
+import io.prestosql.spi.type.Type;
+import org.apache.parquet.column.values.ValuesWriter;
+import org.apache.parquet.schema.OriginalType;
+import org.apache.parquet.schema.PrimitiveType;
+
+import static java.util.Objects.requireNonNull;
+
+public class TimeValueWriter
+        extends PrimitiveValueWriter
+{
+    private final Type type;
+    private final boolean scaleValue;
+
+    public TimeValueWriter(ValuesWriter valuesWriter, Type type, PrimitiveType parquetType)
+    {
+        super(parquetType, valuesWriter);
+        this.type = requireNonNull(type, "type is null");
+        this.scaleValue = parquetType.isPrimitive() && parquetType.getOriginalType() == OriginalType.TIME_MICROS;
```

Comment:
Name the class `TimeMicrosValueWriter` and check the logical type outside, then we don't need to specialize inside the writer code.

Reply:
Done.  It's great to remove that grotty check in the writer!